### PR TITLE
Revise documentation 

### PR DESCRIPTION
**Pull request by CSol314**
Checked spelling, grammar, parameters, and throws- made a few modifications.

Signed-off-by: The MathWorks, Inc. Roy.Lurie@mathworks.com

### CODE REVIEW

File: framework/include/cppmicroservices/Bundle.h [hunk 1]
```diff
@@ -424,7 +424,8 @@ class US_Framework_EXPORT Bundle
    * @return A list of ServiceReference objects for services this
    * bundle has registered.
    *
-   * @throws std::logic_error If this bundle has been uninstalled.
+   * @throws std::logic_error If this bundle has been uninstalled, if 
+   * the ServiceRegistrationBase object is invalid, or if the service is unregistered.
```
Comment (jeffdiclemente) [inline framework/include/cppmicroservices/Bundle.h]: There's an inconsistency with indentation of the newline after the ```@throw``` line. See other instances of ```@throw```

File: framework/include/cppmicroservices/Bundle.h [hunk 2]
```diff
@@ -441,7 +442,8 @@ class US_Framework_EXPORT Bundle
    * @return A list of ServiceReference objects for all services this
    * bundle is using.
    *
-   * @throws std::logic_error If this bundle has been uninstalled.
+   * @throws std::logic_error If this bundle has been uninstalled, if
+   * the ServiceRegistrationBase object is invalid, or if the service is unregistered.
```
Comment (jeffdiclemente) [inline framework/include/cppmicroservices/Bundle.h]: There's an inconsistency with indentation of the newline after the ```@throw``` line. See other instances of ```@throw```

File: framework/include/cppmicroservices/BundleContext.h [hunk 3]
```diff
@@ -343,23 +346,22 @@ class US_Framework_EXPORT BundleContext
...
-   * list contains services that
-   * were registered under the specified class and match the specified filter
-   * expression.
+   * list contains services that were registered under the specified class
+   * and match the specified filter expression.
    *
    * <p>
-   * The list is valid at the time of the call to this method. However since
+   * The list is valid at the time of the call to this method. However, since
    * the Micro Services framework is a very dynamic environment, services can be modified or
```
Comment (jeffdiclemente) [inline framework/include/cppmicroservices/BundleContext.h]: There's an inconsistency. A previous usage of ```Micro Services framework``` was replaced with ```framework```. This usage probably needs to be renamed appropriately.

File: framework/include/cppmicroservices/BundleContext.h [hunk 4]
```diff
@@ -259,12 +265,9 @@ class US_Framework_EXPORT BundleContext
    *         registering the service to update the service's properties or to
    *         unregister the service.
    *
-   * @throws std::invalid_argument If one of the following is true:
-   *         <ul>
-   *         <li><code>service</code> is <code>0</code>.
-   *         <li><code>properties</code> contains case variants of the same key name.
-   *         </ul>
-   * @throws std::logic_error If this BundleContext is no longer valid.
+   * @throws std::runtime_error If this BundleContext is no longer valid.
```
Comment (saschazelzer) [inline framework/include/cppmicroservices/BundleContext.h]: A `std::runtime_error` will also be thrown if there are case variants of the same key in the supplied ` properties` map.

**Pull request merged**